.github/workflows/build.yml

```
@@ -316,6 +316,8 @@ jobs:
       run: |
         if [[ "${{ github.ref }}" == "refs/heads/2.3" ]] && [[ "${{ github.repository }}" == "mixxxdj/mixxx" ]]; then
           CPACK_ARGS="-D DEB_UPLOAD_PPA=ppa:mixxx/mixxxbetas"
+        elif [[ "${{ github.ref }}" =~ ^refs/tags/[0-9]+\.[0-9]+\.[0-9]+$ ]] && [[ "${{ github.repository }}" == "mixxxdj/mixxx" ]]; then
```

Do we need to escape the forward slashes? 
```suggestion
        elif [[ "${{ github.ref }}" =~ ^refs\/tags\/[0-9]+\.[0-9]+\.[0-9]+$ ]] && [[ "${{ github.repository }}" == "mixxxdj/mixxx" ]]; then
```
The line above needs also be fixed.

---

I don't think so. Usually you only need to escape forward slashes for `sed`, because the forward slash is used as separator in `cmd/pattern/repl/flags`. You can actually avoid it even for sed by using a different separator, e.g. `sed 's|/some/path|/other/path|g'` instead of `sed 's/\/some\/path/\/other\/path/g'`.

You can just test it locally: https://github.com/mixxxdj/mixxx/pull/3800#issuecomment-823242744

---

https://www.regextester.com/ confirms that it is working, but it warns about the issue. 
Does this warning apply for us?

---

That website only supports PCRE and JavaScript regexes, which always use "/" as regex separator (you can see this because the input field is enclosed in `/`). So in that case you'd need to escape `/` of course. On Regex101 it does also warn when using PCRE/JS, but not for Python/Golang/Java. However, none of those apply to us as bash uses another dialect.

From the bash man page:

> An additional binary operator, =~, is available, with the same precedence as == and !=.  When it is used, the string to the right of the operator is considered a POSIX extended regular expression and matched accordingly (using  the  POSIX  regcomp  and regexec  interfaces  usually  described  in regex(3)).

`man 3 regex` only contains API information, but `man 7 regex` contains information about the regex dialect. The forward slash `/` isn't mentioned anywhere on that man page, so I suppose it's not a special character and does not need to escape.

For testing, I suggest to verify it locally by copy&pasting the command into a bash shell:

    $ [[ "refs/tags/2.4.0" =~ ^refs/tags/[0-9]+\.[0-9]+\.[0-9]+$ ]] ; echo $?
0
    $ [[ "refs/tags/2.4.0-alpha" =~ ^refs/tags/[0-9]+\.[0-9]+\.[0-9]+$ ]] ; echo $?
1

---

It works, as well as the escaped version.